contracts/GameWallet/GameWallet.sol

```diff
@@ -0,0 +1,221 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+/**
+ * @title GameWallet
+ * @dev A contract for managing participants' token balances and distributing prizes.
+ */
+contract GameWallet is OwnableUpgradeable {
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    // Address of prize token
+    IERC20Upgradeable public prizeToken;
+
+    // Mapping for users balance
+    mapping(address => uint256) public pBalance;
+
+    // Mapping for withdraw lock
+    mapping(address => uint256) public lockUntil;
```

Comment:
Remove lockUntil mapping here and move to under lockDuration variable definition.
And leave mapping(address => bool) public locked definition here as its storage was already used.